src/synapse/workload.py

```diff
@@ -325,6 +325,28 @@ def enable_serve_server_wellknown(container: ops.Container) -> None:
         raise WorkloadError(str(exc)) from exc
 
 
+def enable_federation_domain_whitelist(container: ops.Container, charm_state: CharmState) -> None:
+    """Change the Synapse configuration to enable federation_domain_whitelist.
+
+    Args:
+        container: Container of the charm.
+        charm_state: Instance of CharmState.
+
+    Raises:
+        WorkloadError: something went wrong enabling configuration.
+    """
+    try:
+        config = container.pull(SYNAPSE_CONFIG_PATH).read()
+        current_yaml = yaml.safe_load(config)
+        if charm_state.synapse_config.federation_domain_whitelist is not None:
+            current_yaml[
+                "federation_domain_whitelist"
+            ] = charm_state.synapse_config.federation_domain_whitelist.split(",")
```

Comment:
nit: maybe we can trim each individual items from the comma-separated list to allow users to insert whitespace or newline after comma for better clarity. Like `example.com, example.net`. (feel free to merge without this as this is not vert necessary)

Reply:
Nice suggestion, thanks, I'll add this to be changed in a follow-up PR